crucible-llvm/src/Lang/Crucible/LLVM/MemModel.hs

```diff
@@ -776,11 +776,13 @@ doFree sym mem ptr = do
          Nothing -> memImplHandleMap mem
 
   -- NB: free is defined and has no effect if passed a null pointer
-  isNull <- ptrIsNull sym PtrWidth ptr
-  p1'    <- orPred sym p1 isNull
-  p2'    <- orPred sym p2 isNull
+  isNull   <- ptrIsNull sym PtrWidth ptr
+  p1'      <- orPred sym p1 isNull
+  p2'      <- orPred sym p2 isNull
+  notFreed <- notPred sym wasFreed
   assertUndefined sym p1' (UB.FreeBadOffset (RV ptr))
   assertUndefined sym p2' (UB.FreeUnallocated (RV ptr))
+  assertUndefined sym notFreed (UB.DoubleFree (RV ptr))
```

Comment:
If I've understood the semantics of this correctly, you will get an assertion from both `FreeUnallocated` and `DoubleFree` in the double-free case, right? Maybe you only want to assert `p2' OR wasFreed` in the `FreeUnallocated` case to make them disjoint?

Reply:
Changed to disjoint.